test/e2e/os/testdata/nvidia-driver-install.sh

```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
+set -e
+
+# Detect if NVIDIA GPU is present
+if ! lspci | grep -i nvidia > /dev/null; then
+    echo "No NVIDIA GPU detected, skipping driver installation"
+    exit 0
+fi
+
+echo "NVIDIA GPU detected, installing drivers"
+
+# Detect architecture
+ARCH=$(uname -m)
+if [[ "$ARCH" == "x86_64" ]]; then
+    ARCH_DIR_NVIDIA="x86_64"
+    ARCH_NAME="amd64"
+elif [[ "$ARCH" == "aarch64" ]]; then
+    ARCH_DIR_NVIDIA="sbsa"
+    ARCH_NAME="arm64"
+else
+    echo "Unsupported architecture: $ARCH"
+    exit 1
+fi
+
+echo "Detected architecture: $ARCH_NAME"
+
+# Detect OS type
+OS_TYPE=""
+PKG_TYPE=""
+if [ -f /etc/os-release ]; then
+    source /etc/os-release
+    if [[ "$ID" == "amzn" ]]; then
+        OS_TYPE="al23"
+        PKG_TYPE="rpm"
+    elif [[ "$ID" == "ubuntu" ]]; then
+        OS_TYPE="ubuntu"
+        PKG_TYPE="deb"
+    elif [[ "$ID" == "rhel" ]]; then
+        OS_TYPE="rhel"
+        PKG_TYPE="rpm"
+    fi
+fi
+
+if [ -z "$OS_TYPE" ]; then
+    echo "Unsupported OS type, cannot install NVIDIA drivers"
+    exit 1
+fi
+
+# Install drivers based on OS type
+case $OS_TYPE in
+    ubuntu)
+        echo "Installing NVIDIA drivers for Ubuntu ($ARCH_NAME)"
+        # https://docs.nvidia.com/datacenter/tesla/driver-installation-guide/index.html#ubuntu-installation
+        # Install required packages
+        apt-get update
+        apt-get install -y linux-headers-$(uname -r)
```

Comment:
Should we quote this and every time we substitute this to prevent word splitting? 
```suggestion
        apt-get install -y "linux-headers-$(uname -r)"
```

Reply:
that should be okay. I just extract `uname -r` into a variable.